src/v/storage/ntp_config.h

```diff
@@ -183,6 +185,13 @@ class ntp_config {
             }
             // If no value set, fall through and use the cluster-wide default.
         }
+
+        if (is_read_replica_mode_enabled()) {
+            // Read replicas have a special hardcoded default, because they do
+            // not retain user data in local raft log, just configuration.
+            return read_replica_retention;
+        }
+
```

Comment:
Is this meant to be a stop-gap until read-replica max collectible offset is fixed? If max-collectible is set to the last snapshoted index, I think we could bump these to retain nothing, right?

Reply:
I think setting it to zero is already safe, but one has to pick some finite time to set segment.ms, so we might as well not try and trim any more aggressively than we're rolling segments.